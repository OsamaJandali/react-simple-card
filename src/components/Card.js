// @flow

import React from "react";

type CardProps = {
	children: any,
	style?: any,
	className?: string
};

const defaultStyles = {
	width: "100%",
	display: "block",
	position: "relative",
	backgroundColor: "#fff",
	borderRadius: "4px",
	marginBottom: "10px",
	border: "1px solid #ccc"
};

export default ({ className = "", style, children }: CardProps) => {
	const styles = { ...defaultStyles, ...style };
	return (
		<div className={`card ${className}`} style={styles}>
			{children}
		</div>
	);
};
