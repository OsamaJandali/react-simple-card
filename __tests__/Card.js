// @flow

import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Card from "../src/components/Card";

it("renders Card correctly with default styles", () => {
	const wrapper = shallow(
		<Card>Aliquam sodales risus ex, sit amet posuere magna interdum eget.</Card>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders Card correctly with extra styles", () => {
	const wrapper = shallow(
		<Card style={{ minHeight: "100px" }}>
			Aliquam sodales risus ex, sit amet posuere magna interdum eget.
		</Card>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders Card correctly with black background color", () => {
	const wrapper = shallow(
		<Card bgColor="#000">
			Aliquam sodales risus ex, sit amet posuere magna interdum eget.
		</Card>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders Card correctly with blue font style", () => {
	const wrapper = shallow(
		<Card fontColor="blue">
			Aliquam sodales risus ex, sit amet posuere magna interdum eget.
		</Card>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});
