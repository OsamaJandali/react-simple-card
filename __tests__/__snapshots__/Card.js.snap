// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<div
  className="card "
  style={
    Object {
      "backgroundColor": "#fff",
      "border": "1px solid #ccc",
      "borderRadius": "4px",
      "display": "block",
      "marginBottom": "10px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  Aliquam sodales risus ex, sit amet posuere magna interdum eget.
</div>
`;
